jobserver/authorization/roles.py

```diff
@@ -0,0 +1,72 @@
+from .permissions import (
+    cancel_job,
+    check_output,
+    invite_project_members,
+    publish_output,
+    review_project,
+    run_job,
+)
+
+
+class CoreDeveloper:
```

Comment:
Add a docstring here explaining the meaning of each role?

This should then be included in our official documentation, I think. I realise most of the roles won't mean anything much *yet* but I think freezing our terminology is probably sensible, i.e. copying from that Google Doc, adding a highlighted note there to say it's now in the docs.

That also implies adding a new page to the docs which imports the docstrings. That part could be a relatively straightfoward job for @StevenMaude who's starting out by owning our new user experience?

Reply:
In the platform docs or docs for this project?

I'm not sure we should document the implementation of Roles as part of the platform docs?  Importing a module from this project (assuming that means we have to import the whole project) feels like a very heavy weight way of pulling anything into the docs (which already have to deal with the cohort-extractor's dependencies), and I'm not keen on making the docs project harder to build.

Reply:
Oh I forgot we're not in cohortextractor! Maybe that's not the way - but we do need to freeze "description of roles" somewhere and link to it from here - so people can make decisions about which roles/permissions to use for new features, etc.

Reply:
Added docstrings to the Roles, rebased changes, updated doc.

Reply:
I've not really tidied up the descriptions too much as I think they're pretty descriptive as they are but we might want to do a pass later.